cli/client.go

```diff
@@ -380,6 +387,123 @@ func RobotsPartShellAction(c *cli.Context) error {
 	)
 }
 
+// checkUpdateResponse holds the values used to hold release information.
+type checkUpdateResponse struct {
+	Name       string `json:"name"`
+	TagName    string `json:"tag_name"`
+	TarballURL string `json:"tarball_url"`
+}
+
+func checkLatestRelease() (checkUpdateResponse, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+
+	resp := checkUpdateResponse{}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, rdkReleaseURL, nil)
+	if err != nil {
+		return resp, err
+	}
+
+	client := http.DefaultClient
+	res, err := client.Do(req)
+	if err != nil {
+		return resp, err
+	}
+
+	err = json.NewDecoder(res.Body).Decode(&resp)
+	if err != nil {
+		return resp, err
+	}
+
+	defer utils.UncheckedError(res.Body.Close())
+	return resp, err
+}
+
+// CheckUpdateAction is the corresponding Action for 'check-update'.
+func CheckUpdateAction(c *cli.Context) error {
+	if c.Bool(quietFlag) {
+		return nil
+	}
+
+	dateCompiledRaw := rconfig.DateCompiled
+	dateCompiled, err := time.Parse("2006-01-02", dateCompiledRaw)
+	if err != nil {
+		warningf(c.App.ErrWriter, "CLI Update Check: failed to parse compilation date: %w", err)
```

Comment:
On your branch, the output of my `go run ./cli/viam/main.go login` outputs: `Warning: CLI Update Check: failed to parse compilation date: %!w(*time.ParseError=&{2006-01-02  2006  })`. Seems like you have to run `make cli` to get the LDFLAGS to fill in `DateCompiled` and avoid this warning . Maybe we should just return `nil` if `dateCompiledRaw == ""` since we have no information on when the CLI was compiled?